xbmc/interfaces/legacy/Addon.cpp

```diff
@@ -81,9 +81,61 @@ namespace XBMCAddon
       return g_localizeStrings.GetAddonString(pAddon->ID(), id);
     }
 
-    Settings* Addon::getSettings()
+    bool Addon::supportsInstanceSettings()
     {
-      return new Settings(pAddon->GetSettings());
+      return pAddon->SupportsInstanceSettings();
+    }
+
+    std::vector<unsigned int> Addon::getKnownInstanceIds()
+    {
+      return pAddon->GetKnownInstanceIds();
+    }
+
+    unsigned int Addon::getFreeNewInstanceId()
+    {
+      const auto ids = pAddon->GetKnownInstanceIds();
+      ADDON::AddonInstanceId highestId = ADDON::ADDON_FIRST_SCRIPT_SET_INSTANCE_ID;
+      for (const auto& id : ids)
+      {
+        if (id == ADDON::ADDON_SINGLETON_INSTANCE_ID)
+          throw XBMCAddon::WrongTypeException("Invalid add-on type");
+
+        if (id >= ADDON::ADDON_FIRST_SCRIPT_SET_INSTANCE_ID && id > highestId)
+          highestId = id;
+      }
+
+      return highestId + 1;
+    }
+
+    bool Addon::deleteInstanceId(unsigned int instance)
+    {
+      bool ret = pAddon->DeleteInstanceSettings(instance);
+      if (ret)
+        CServiceBroker::GetAddonMgr().PublishInstanceRemoved(pAddon->ID(), instance);
+      return ret;
+    }
+
+    bool Addon::publishInstanceChange(unsigned int instance)
+    {
+      if (pAddon->SaveSettings(instance))
+      {
+        bool value = false;
+        if (!pAddon->GetSettingBool("kodi_addon_instance_enabled", value, instance))
+          throw XBMCAddon::WrongTypeException("Invalid setting type");
+
+        if (value)
+          CServiceBroker::GetAddonMgr().PublishInstanceAdded(pAddon->ID(), instance);
```

Comment:
Will this callback to the PVR add-on? It is important that it does as otherwise the add-on cannot handle settings changes. Ideally each changed setting would callback to `SetInstanceSetting()` for the add-on implementing the interface kodi::addon::IAddonInstance.

Reply:
> Will this callback to the PVR add-on? It is important that it does as otherwise the add-on cannot handle settings changes. Ideally each changed setting would callback to `SetInstanceSetting()` for the add-on implementing the interface kodi::addon::IAddonInstance.
> 
> IT's also possible this happens for free as part of the regular settings interface.

The callback mainly works to the CPVRClients, where then call the PVR add-ons and only needed on add or disable/enable. `SetInstanceSetting()` now becomes called if instance already know on add-on.

Reply:
@phunkyfish 
i dont have the env setup to create builds sorry. If you can get me an x86_64 windows I can test.
We want to make sure that we can force IPTV Simple to reload its playlists right?

Reply:
> @phunkyfish i dont have the env setup to create builds sorry. If you can get me an x86_64 windows I can test. We want to make sure that we can force IPTV Simple to reload its playlists right?

This mainly to change the use state about the instance, e.g. the disable/enable. Normally should the add-on handle setting changes by self, if he see a change within in his settings. But optionally can the python also do a hacky disable and enable again to force a reload.